src/main/java/julia/git/Directory.java

```diff
@@ -0,0 +1,69 @@
+package julia.git;
+
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class Directory implements Listable {
+    private final HashMap<String, Listable> content = new HashMap<>();
+
+    public void add(String name, Listable list) {
+        content.put(name, list);
+    }
+
+    public List<String> list() {
+        return new ArrayList<>(content.keySet());
+    }
+
+    public Listable get(String name) {
+        return content.get(name);
+    }
+
+    @Override
+    public boolean isDirectory() {
+        return true;
+    }
+
+    @Override
+    public String getContent() throws UnsupportedEncodingException, NoSuchAlgorithmException {
+        StringBuilder builder = new StringBuilder();
+        for (String key : content.keySet()) {
+            Listable list = content.get(key);
+            String hash = list.getHash();
+            builder.append(hash).append("     ").append(key).append("\n");
+        }
+        builder.deleteCharAt(builder.length() - 1);
+        return builder.toString(); // Get string containing keys + hashes
+    }
+
+    @Override
+    public String getHash() throws NoSuchAlgorithmException {
+        StringBuilder builder = new StringBuilder();
+        for (String key : content.keySet()) {
```

Comment:
This one is still under construction? We should use `getContent()` instead of constructing the string ourselves.